keras/engine/topology.py

```diff
@@ -2290,18 +2308,21 @@ def get_config(self):
             else:
                 kept_nodes = 0
             for original_node_index, node in enumerate(layer.inbound_nodes):
-                node_key = layer.name + '_ib-' + str(original_node_index)
+                node_key = self._node_key(layer, original_node_index)
                 if node_key in self.container_nodes:
+                    # i.e. we mark it to be saved
                     node_conversion_map[node_key] = kept_nodes
                     kept_nodes += 1
+
+        # serialize and save the layers in layer_configs
         layer_configs = []
         for layer in self.layers:  # From the earliest layers on.
             layer_class_name = layer.__class__.__name__
             layer_config = layer.get_config()
             filtered_inbound_nodes = []
             for original_node_index, node in enumerate(layer.inbound_nodes):
-                node_key = layer.name + '_ib-' + str(original_node_index)
-                if node_key in self.container_nodes:
+                if (self._node_key(layer, original_node_index) in
+                        self.container_nodes):
```

Comment:
Style nit: avoid strange line breaking

```python
node_key = self._node_key(layer, original_node_index)
if node_key in self.container_nodes:
```

Also applicable in several other places in this PR. Please fix.